File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{
    //float quarter, dime, nickel, penny;
    int coins = 0;
    
    printf("How much change is owed?\n");
    float money = get_float();
    while(money < 0) //Checks that the amount is a positive Float
    {
        printf("How much change is owed?\n");
        money = get_float();
    }
    int cents = round(money * 100); //Converts the money to cents to avoid floats imprecisions
    
    
    while (cents >= 25) //quarters
    {
        coins++;
        cents = cents - 25;
    }
    while (cents >= 10) //dimes
    {
        coins++;
        cents = cents - 10;
    }
    while (cents >= 5) //nickels
    {
        coins++;
        cents = cents - 5;
    }
    while (cents >= 1) //pennies
    {
        coins++;
        cents = cents - 1;
    }    
    
    printf("%i\n", coins);
}

File: pset6/greedy.py
def main():

	while True:
		try:
			cash = float(input("Amount: "))
		except ValueError:
			continue
		else:
			if cash > 0 and cash < 100:
				break
			else:
				continue

	quarter, dime, nickle, penny = 0,0,0,0

	while cash >= 0.25:
		cash -= 0.25
		quarter += 1

	while cash >= 0.10:
		cash -= 0.10
		dime += 1

	while cash >= 0.05:
	    cash -= 0.05
	    nickle += 1

	while cash >= 0.01:
		cash -= 0.01
		penny += 1

	print(quarter)
	print(dime)
	print(nickle)
	print(penny)


if __name__ == "__main__":
	main()
